app/controllers/mixins/generic_show_mixin.rb

```diff
@@ -107,6 +107,26 @@ def display_nested_generic(display)
       nested_list(display, display.camelize.singularize.constantize)
     end
 
+    def display_descendant_vms
+      @showtype = "config"
+      drop_breadcrumb(:name => _("%{name} (All VMs - Tree View)") % {:name => @record.name},
+                      :url  => "/#{controller_name}/show/#{@record.id}?display=descendant_vms&treestate=true")
```

Comment:
Also I wonder if the `display_descendant_vms` here is going to be used by many controllers or just by one or a few. In the latter case I would not put it into the mixin. I'd put it in the controller that calls it or into a mixin that would be included in the controllers tha use it.